index.js

```diff
@@ -50,6 +56,10 @@ async function fastifyStatic (fastify, opts) {
       }
     }
 
+    if (!allowedPath(pathname, options.root)) {
```

Comment:
With this check, it would be possible to skip the function declaration and the function call that returns `true`
```suggestion
    if (allowedPath && !allowedPath(pathname, options.root)) {
```

Reply:
Done